frontend/src/component/ui/Table.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component, ReactChild, ReactText } from 'react';
+
+import './Table.scss';
+
+interface IProps {
+  headings?: ReactChild[];
+  rows?: ReactChild[][];
+}
+
+export class Table extends Component<IProps> {
+  private constructHeadIfExists(headings: ReactChild[] | undefined) {
+    if (!headings || headings.length === 0) {
+      return null;
+    }
+    return <tr>{this.constructRow(headings, true)}</tr>;
+  }
+
+  private constructRow(row: ReactChild[], isHeader: boolean) {
+    return row.map((cell: ReactChild, cellIndex: number) => {
+      return isHeader
+        ? this.constructHeadingCell(cellIndex, cell)
+        : this.constructBodyCell(cellIndex, cell);
+    });
+  }
+
+  private constructHeadingCell(key: ReactText, content: ReactChild) {
+    return (
+      <th key={`${key}`} className="table-cell">
+        {content}
+      </th>
+    );
+  }
+
+  private constructBodyCell(key: ReactText, content: ReactChild) {
+    return (
+      <td key={`${key}`} className="table-cell">
+        {content}
+      </td>
+    );
+  }
+
+  private constructBodyIfExists(rows: ReactChild[][] | undefined) {
+    if (!rows || rows.length === 0) {
+      return null;
+    }
+    return rows.map((row: ReactChild[], rowIndex: number) => {
+      return <tr key={`row-${rowIndex}`}>{this.constructRow(row, false)}</tr>;
```

Comment:
```suggestion
      return (
      <tr key={`row-${rowIndex}`}>
          {this.createRow(row)}
      </tr>
      );
```

Reply:
Same, prettier doesn't allow to split across lines if the number of characters are less than 80 when it involves typescript code. It allows the following line, but not this.
```
    return rows.map((row: ReactChild[], rowIndex: number) => {
      return (
        <tr key={`row-${rowIndex}`}>
          <div>hello</div>
        </tr>
      );
    });
```